packages/react-textarea/Spec.md

```diff
@@ -2,62 +2,174 @@
+- Has descriptions, label, error message, and handles maximum character count.
+
+#### TextArea in v0
+
+- Doesn't have `label` prop.
+- Handles maximum character count.
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```tsx
+<TextArea
+  className="rootClassName"
+  style={{ background: 'blue' }}
+  id="textarea-1"
+  defaultValue="This is a TextArea"
+  onChange={(ev, data) => console.log(data.value)}
+  textArea={{ className: 'textAreaClassName', style: { color: 'green' } }}
+/>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+- TextArea has six variants:
+  - `Filled Darker`: TextArea has a gray background and no outline.
+  - `Filled Lighter`: TextArea has a white background and no outline.
+  - `Underline`: TextArea has a gray background and a line on the bottom.
+  - `Outline`: TextArea has a white background, outline, and a line on the bottom.
+  - `Disabled`: TextArea has a gray background, gray text, outline, and interaction is disabled.
+  - `Read Only`: TextArea has a gray background, gray outline, and typing is disabled.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### Component props:
+
+```ts
+type TextAreaProps = {
+  /**
+   * Display inline instead of block.
+   *
+   * @defaultvalue false
+   */
+  inline?: boolean;
+
+  /**
+   * Which direction the TextArea is allowed to be resized.
+   *
+   * @defaultvalue off
+   */
+  resize?: 'off' | 'horizontal' | 'vertical' | 'both';
+
+  /**
+   * Direction the TextArea should resize.
+   *
+   * @defaultvalue off
+   */
+  autoResize?: 'off' | 'top' | 'bottom';
+
+  /**
+   * Size of the TextArea.
+   *
+   * @defaultvalue medium
+   */
+  size?: 'small' | 'medium' | 'large';
+
+  /**
+   * Styling the TextArea should use.
+   *
+   * @defaultvalue outline
+   */
+  appearance?: 'outline' | 'underline' | 'filledDarker' | 'filledLighter';
+
+  /**
+   * The value of the TextArea.
+   */
+  value?: string;
+
+  /**
+   * The default value of the TextArea.
+   */
+  defaultValue?: string;
+
+  /**
+   * Callback for when the user changes the value.
+   */
+  onChange?: (ev: React.FormEvent<HTMLTextAreaElement>, data: { value: string }) => void;
+};
+```
+
+As with most form controls, `TextArea` will have the `textarea` element as the primary element a wrapper (root). This means that `ref` will be passed directly to the `textarea` with its native props.
+
+The design spec mentions a character count, error message, and label which will be left out for now.
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+**Public**
 
-## Migration
+```tsx
+<TextArea
+  className="rootClassName"
```

Comment:
I think this will need some work because the `textArea` slot is the primary slot so className will be applied to it which is different from your proposed DOM output below.